src/mc-fle2-payload-iev-v2.c

```diff
@@ -322,9 +325,93 @@ void mc_FLE2IndexedEncryptedValueV2_destroy(mc_FLE2IndexedEncryptedValueV2_t *ie
     _mongocrypt_buffer_cleanup(&iev->DecryptedServerEncryptedValue);
     _mongocrypt_buffer_cleanup(&iev->ServerEncryptedValue);
     _mongocrypt_buffer_cleanup(&iev->S_KeyId);
+
+    if (iev->type != kTypeRange) {
+        for (int i = 0; i < iev->edge_count; i++) {
+            mc_FLE2TagAndEncryptedMetadataBlock_cleanup(&iev->metadata[i]);
+        }
+    } else {
+        // kTypeEquality only has one metadata block
+        mc_FLE2TagAndEncryptedMetadataBlock_cleanup(iev->metadata);
+    }
+
+    // Metadata array is dynamically allocated
+    bson_free(iev->metadata);
+
     bson_free(iev);
 }
 
+uint8_t mc_FLE2IndexedEncryptedValueV2_get_edge_count(const mc_FLE2IndexedEncryptedValueV2_t *iev,
+                                                      mongocrypt_status_t *status) {
+    assert(iev);
+
+    if (iev->type == kTypeInit) {
+        CLIENT_ERR("mc_FLE2IndexedEncryptedValueV2_get_edge_count "
+                   "must be called after "
+                   "mc_FLE2IndexedEncryptedValueV2_parse");
+        return 0;
+    }
+
+    if (iev->type != kTypeRange) {
+        CLIENT_ERR("mc_FLE2IndexedEncryptedValueV2_get_edge_count must be called with type range");
+        return 0;
```

Comment:
Equality has precisely one edge, so what if we simply set `iev->edge_count = 1;` in the parse method for the equality version and then let these methods treat range and equality the same?